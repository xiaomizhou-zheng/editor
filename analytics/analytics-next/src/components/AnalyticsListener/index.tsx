import LegacyAnalyticsListener from './LegacyAnalyticsListener';
import ModernAnalyticsListener from './ModernAnalyticsListener';
import { AnalyticsListenerFunction } from './types';

let ExportedAnalyticsListener: AnalyticsListenerFunction;
if (
  typeof process !== 'undefined' &&
  process !== null &&
  process.env?.['ANALYTICS_NEXT_MODERN_CONTEXT']
) {
  ExportedAnalyticsListener = ModernAnalyticsListener as any;
} else {
  ExportedAnalyticsListener = LegacyAnalyticsListener as any;
}

export default ExportedAnalyticsListener;
