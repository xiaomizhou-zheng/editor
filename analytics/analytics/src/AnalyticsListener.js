/* eslint-disable react/sort-comp */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import matchEvent from './matchEvent';

/*
The Listener component is responsible for calling its `onEvent` handler when a
child component fires an analytics event, and passing the event up the hierarchy
*/
const ContextTypes = {
  onAnalyticsEvent: PropTypes.func,
};

if (process.env.NODE_ENV !== 'production' && !process.env.CI) {
  // eslint-disable-next-line no-console
  console.warn(
    'The @atlaskit/analytics package has been deprecated. Please use the @atlaskit/analytics-next package instead.',
  );
}

class AnalyticsListener extends Component {
  static defaultProps = {
    match: '*',
    matchPrivate: false,
  };

  static contextTypes = ContextTypes;

  static childContextTypes = ContextTypes;

  getChildContext() {
    return {
      onAnalyticsEvent: this.onAnalyticsEvent,
    };
  }

  onAnalyticsEvent = (name, data, isPrivate) => {
    // Call this component's onEvent method if it's a match
    // eslint-disable-next-line react/prop-types
    const { onEvent, match, matchPrivate } = this.props;
    if (
      matchPrivate === isPrivate &&
      matchEvent(match, name) &&
      typeof onEvent === 'function'
    ) {
      // send a clean data object so it can't be mutated between listeners
      const eventData = { ...data };
      onEvent(name, eventData);
    }

    // Pass the event up the hierarchy
    const { onAnalyticsEvent } = this.context;
    if (typeof onAnalyticsEvent === 'function') {
      onAnalyticsEvent(name, data, isPrivate);
    }
  };

  render() {
    const { children } = this.props; // eslint-disable-line react/prop-types
    return React.Children.only(children);
  }
}

export default AnalyticsListener;
