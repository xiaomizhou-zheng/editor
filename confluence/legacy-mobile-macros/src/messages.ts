import { defineMessages } from 'react-intl-next';

import {
  ERROR_LOADING_TEXT,
  FINAL_ERROR_LOADING_TEXT,
  TAP_TO_LOAD_TEXT,
  TAP_TO_REFRESH_PAGE_TEXT,
  TAP_TO_RETRY_TEXT,
  TAP_TO_VIEW_TEXT,
} from './ui/MacroComponent/MacroFallbackComponent/constants';

export const legacyMobileMacrosMessages = defineMessages({
  tapToLoadMacro: {
    id: 'fabric.editor.confluence.legacyMobileMacros.tapToLoadMacro',
    defaultMessage: TAP_TO_LOAD_TEXT,
    description: 'Tap block to load macro',
  },
  tapToViewMacro: {
    id: 'fabric.editor.confluence.legacyMobileMacros.tapToViewMacroMacro',
    defaultMessage: TAP_TO_VIEW_TEXT,
    description: 'Tap block to view macro',
  },
  tapToRetryLoadingMacro: {
    id: 'fabric.editor.confluence.legacyMobileMacros.tapToRetryLoadingMacro',
    defaultMessage: TAP_TO_RETRY_TEXT,
    description: 'Tap to retry loading macro after error',
  },
  errorLoadingMacro: {
    id: 'fabric.editor.confluence.legacyMobileMacros.errorLoadingMacro',
    defaultMessage: ERROR_LOADING_TEXT,
    description: 'Error loading macro',
  },
  finalErrorLoadingMacro: {
    id: 'fabric.editor.confluence.legacyMobileMacros.finalErrorLoadingMacro',
    defaultMessage: FINAL_ERROR_LOADING_TEXT,
    description: 'Could not load macro after final retry',
  },
  tapToRefreshPage: {
    id: 'fabric.editor.confluence.legacyMobileMacros.tapToRefreshPage',
    defaultMessage: TAP_TO_REFRESH_PAGE_TEXT,
    description: 'Tap block to refresh the macro view page',
  },
});
