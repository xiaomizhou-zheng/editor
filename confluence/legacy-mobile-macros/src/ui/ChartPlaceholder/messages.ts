import { defineMessages } from 'react-intl-next';

export const chartPlaceholderMessages = defineMessages({
  chartPlaceholderText: {
    id: 'fabric.editor.confluence.legacyMobileMacros.chartPlaceholderText',
    defaultMessage:
      'A chart should be here, but it canâ€™t be shown. Open this page in a browser to see the chart.',
    description:
      'Message shown in place of chart extension on mobile (where the extension is not yet supported)',
  },
});
