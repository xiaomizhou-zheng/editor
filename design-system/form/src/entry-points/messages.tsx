export { HelperMessage, ErrorMessage, ValidMessage } from '../messages';
