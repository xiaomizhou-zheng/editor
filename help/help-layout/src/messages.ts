import { defineMessages } from 'react-intl-next';

export const messages = defineMessages({
  help_loading: {
    id: 'helpPanel.loading',
    defaultMessage: 'Loading',
    description: '',
  },
  help_panel_header_title: {
    id: 'helpPanel.header.title',
    defaultMessage: 'Help',
    description: '',
  },
  help_panel_header_back: {
    id: 'helpPanel.header.back',
    defaultMessage: 'Back',
    description: '',
  },
  help_panel_header_close: {
    id: 'helpPanel.header.close',
    defaultMessage: 'Close',
    description: '',
  },
});
