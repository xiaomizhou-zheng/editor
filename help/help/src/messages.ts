import { defineMessages } from 'react-intl-next';

export const messages = defineMessages({
  help_loading: {
    id: 'help.loading',
    defaultMessage: 'Loading',
    description: 'Loading text',
  },
  help_header: {
    id: 'help.header',
    defaultMessage: 'Help',
    description: 'Header text',
  },
  help_header_whats_new: {
    id: 'help.header_whats_new',
    defaultMessage: `What's new`,
    description: `Header text in "What's New" mode`,
  },
  help_close: {
    id: 'help.close',
    defaultMessage: 'Close',
    description: 'Close text',
  },
  help_navigation_back: {
    id: 'help.navigation.back',
    defaultMessage: 'Back',
    description: 'Back text',
  },
  help_search_placeholder: {
    id: 'help.search.placeholder',
    defaultMessage: 'Search help articles',
    description: 'Place holder text for search input',
  },
  help_article_rating_title: {
    id: 'help.article_rating.title',
    defaultMessage: 'Was this helpful?',
    description: 'Rating article title text',
  },
  help_article_rating_option_yes: {
    id: 'help.article_rating.option.yes',
    defaultMessage: 'Yes',
    description: '"Yes" text used in the article rating options',
  },
  help_article_rating_option_no: {
    id: 'help.article_rating.option.no',
    defaultMessage: 'No',
    description: '"No" text used in the article rating options',
  },
  help_article_rating_form_title: {
    id: 'help.article_rating.form.title',
    defaultMessage: 'Tell us why you gave this rating',
    description: 'Title for the article rating form',
  },
  help_article_rating_accurate: {
    id: 'help.article_rating.accurate',
    defaultMessage: `It isn't accurate`,
    description: `Text for the "isn't accurate" option of the article rating form`,
  },
  help_article_rating_clear: {
    id: 'help.article_rating.clear',
    defaultMessage: `It isn't clear`,
    description: `Text for the "isn't clear" option of the article rating form`,
  },
  help_article_rating_relevant: {
    id: 'help.article_rating.relevant',
    defaultMessage: `It isn't relevant`,
    description: `Text for the "isn't relevant" option of the article rating form`,
  },
  help_article_rating_form_submit: {
    id: 'help.article_rating.form.submit',
    defaultMessage: `Submit`,
    description: `Text for the submit button in the article rating form`,
  },
  help_article_rating_form_Success: {
    id: 'help.article_rating.form.success',
    defaultMessage: `Thanks!`,
    description: 'Success message for the article rating form',
  },
  help_article_rating_form_failed: {
    id: 'help.article_rating.form.failed',
    defaultMessage: `We couldn't submit your feedback.`,
    description: 'Error message for the article rating form',
  },
  help_article_rating_form_failed_try_again: {
    id: 'help.article_rating.form.failed.try_again',
    defaultMessage: `Try again`,
    description: `Text for the "Try again" button of the article rating form`,
  },
  help_article_rating_form_cancel: {
    id: 'help.article_rating.form.cancel',
    defaultMessage: `Cancel`,
    description: `Text for the cancel button in the article rating form`,
  },
  help_article_rating_form_contact_me: {
    id: 'help.article_rating.form.contact_me',
    defaultMessage: `Atlassian can contact me about this feedback`,
    description:
      'Text for the "Contact me" checkbox in the article rating form',
  },
  help_article_list_item_type_whats_new: {
    id: 'help.article_list_item.type.whats_new',
    defaultMessage: `WHAT'S NEW`,
    description: `"What's new" article type title`,
  },
  help_article_list_item_type_help_article: {
    id: 'help.article_list_item.type.help-article',
    defaultMessage: `HELP ARTICLE`,
    description: 'Help article article type title',
  },
  help_related_article_title: {
    id: 'help.related_article.title',
    defaultMessage: 'Related articles',
    description: 'Title of the related articles list',
  },
  help_related_article_endpoint_error_title: {
    id: 'help.related_article.endpoint-error.title',
    defaultMessage: `We're having some trouble`,
    description:
      'Title for the message displayed when the related articles fetching fails',
  },
  help_related_article_endpoint_error_description: {
    id: 'help.related_article.endpoint-error.description',
    defaultMessage: `It's taking us longer than expected to show this content. It's probably a temporary problem.`,
    description:
      'Body for the message displayed when the related articles fetching fails',
  },
  help_related_article_endpoint_error_button_label: {
    id: 'help.related_article.endpoint-error.button-label',
    defaultMessage: `Try Again`,
    description:
      'text for the retry button displayed when the related articles fetching fails',
  },
  help_search_results_search_external_site: {
    id: 'help.search_results.search_external_site',
    defaultMessage: `Can't find what you're looking for? Try again with a different term or `,
    description:
      'Text for the button displayed at the end of the search result used to open an external search site',
  },
  help_search_results_no_results: {
    id: 'help.search_results.no_results',
    defaultMessage: `We can't find anything matching your search`,
    description: `Message displayed when the search doesn't return any results (line 1)`,
  },
  help_search_results_no_results_line_two: {
    id: 'help.search_results.no_results_line_two',
    defaultMessage: `Try again with a different term or `,
    description: `Message displayed when the search doesn't return any results (line 2)`,
  },
  help_search_results_external_site_link: {
    id: 'help.search_results.external_site_link',
    defaultMessage: `search all online help articles.`,
    description: `Text for the button used to open an external search site`,
  },
  help_article_error_title: {
    id: 'help.article_error.title',
    defaultMessage: `Something's gone wrong`,
    description: `Error message title for article loading`,
  },
  help_article_error_text: {
    id: 'help.article_error.text',
    defaultMessage: `Something prevented the page from loading. We will try to reconnect automatically.`,
    description: `Error message body for article loading`,
  },
  help_article_error_button_label: {
    id: 'help.article_error.button_label',
    defaultMessage: `Try now`,
    description:
      'text for the retry button displayed when an article fetching fails',
  },
  help_search_error: {
    id: 'help.search_error',
    defaultMessage: `We are having some trouble`,
    description: `Error message for article search loading (line 1)`,
  },
  help_search_error_line_two: {
    id: 'help.search_error_line_two',
    defaultMessage: `It's taking us longer than expected to show this content. It's probably a temporary problem.`,
    description: `Error message for article search loading (line 2)`,
  },
  help_search_error_button_label: {
    id: 'help.search_error.button_label',
    defaultMessage: `Retry`,
    description: 'text for the retry button displayed when the search fails',
  },
  help_whats_new_button_label: {
    id: 'help.whats_new.button_label',
    defaultMessage: 'Find out what’s changed in {productName}',
    description: `text for the "What's New" button`,
  },
  help_whats_new_button_label_without_product_name: {
    id: 'help.whats_new.button_label',
    defaultMessage: 'Find out what’s changed',
    description: `text for the "What's New" button`,
  },
  help_whats_new_button_tooltip: {
    id: 'help.whats_new.button_tooltip',
    defaultMessage: `See our latest feature announcements, improvements, bug fixes, and more.`,
    description: `text for the "What's New" tooltip`,
  },
  help_whats_new_filter_select_option_all: {
    id: 'help.whats_new.filter_select_option_all',
    defaultMessage: `All release types`,
    description: `text for the "Release types" dropdown`,
  },
  help_whats_new_filter_select_option_new: {
    id: 'help.whats_new.filter_select_option_new',
    defaultMessage: `New`,
    description: `text for the "Release types" dropdown option "New"`,
  },
  help_whats_new_filter_select_option_improvement: {
    id: 'help.whats_new.filter_select_option_improvement',
    defaultMessage: `Improvement`,
    description: `text for the "Release types" dropdown option "Improvement"`,
  },
  help_whats_new_filter_select_option_fix: {
    id: 'help.whats_new.filter_select_option_fix',
    defaultMessage: `Fix`,
    description: `text for the "Release types" dropdown option "Fix"`,
  },
  help_whats_new_filter_select_option_removed: {
    id: 'help.whats_new.filter_select_option_removed',
    defaultMessage: `Removed`,
    description: `text for the "Release types" dropdown option "Removed"`,
  },
  help_whats_new_filter_select_option_experiment: {
    id: 'help.whats_new.filter_select_option_experiment',
    defaultMessage: `Experiment`,
    description: `text for the "Release types" dropdown option "Experiment"`,
  },
  help_whats_new_no_results: {
    id: 'help.whats.new_no_results',
    defaultMessage: `We can’t find any recent releases of that type.`,
    description: `text to display when the "What's new" API returns an empty array of articles"`,
  },
  help_whats_new_no_results_info: {
    id: 'help.whats.new_no_results_info',
    defaultMessage: `We only display three months of recent releases.`,
    description: `text to display when the "What's new" API returns an empty array of articles"`,
  },
  help_whats_new_no_results_clear_filter_button_label: {
    id: 'help.whats.new_no_results_clear_filter_button_label',
    defaultMessage: `Clear the filter`,
    description: `text to display when the "What's new" API returns an empty array of articles"`,
  },
  help_whats_new_no_results_clear_filter_info: {
    id: 'help.whats.new_no_results_clear_filter_button_label',
    defaultMessage: ` to try again.`,
    description: `text to display when the "What's new" API returns an empty array of articles"`,
  },
  help_whats_new_related_link_support: {
    id: 'help.whats_new.related_links.support',
    defaultMessage: `Support documentation`,
    description: `text for the "Support documentation" link in the related links section of "what's new" articles`,
  },
  help_whats_new_related_link_community: {
    id: 'help.whats_new.related_links.community',
    defaultMessage: `Discuss with community`,
    description: `text for the "Discuss with community" link in the related links section of "what's new" articles`,
  },
  help_show_more_button_label_more: {
    id: 'help.show_more_button.label_more',
    defaultMessage: 'Show {numberOfItemsLeft} more {itemsType}',
    description: '"Show more" text',
  },
  help_show_more_button_label_less: {
    id: 'help.show_more_button.label_less',
    defaultMessage: 'Show less',
    description: '"Show less" text',
  },
});
