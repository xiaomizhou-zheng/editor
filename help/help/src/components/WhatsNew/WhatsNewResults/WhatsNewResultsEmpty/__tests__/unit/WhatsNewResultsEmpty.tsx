import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render, fireEvent } from '@testing-library/react';
import AnalyticsListener from '@atlaskit/analytics-next/AnalyticsListener';
import { createIntl, createIntlCache } from 'react-intl-next';

import { messages } from '../../../../../../messages';
import { WhatsNewResultsEmpty } from '../..';

const cache = createIntlCache();
const intl = createIntl(
  {
    locale: 'en',
    messages: {},
  },
  cache,
);
const messageClearFilterLink = intl.formatMessage(
  messages.help_whats_new_no_results_clear_filter_button_label,
);

const mockOnClearFilter = jest.fn();
const analyticsSpy = jest.fn();

describe('WhatsNewResultsEmpty', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <WhatsNewResultsEmpty intl={intl} onClearFilter={mockOnClearFilter} />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Execute the function prop "onClearFilter" when the user clicks the link to open clear the filter', () => {
    const { queryByText } = render(
      <AnalyticsListener channel="help" onEvent={analyticsSpy}>
        <WhatsNewResultsEmpty intl={intl} onClearFilter={mockOnClearFilter} />
      </AnalyticsListener>,
    );

    const LinkLabel = queryByText(messageClearFilterLink);
    expect(LinkLabel).not.toBeNull();

    if (LinkLabel) {
      const link = LinkLabel.closest('a');

      expect(link).not.toBeNull();

      if (link) {
        expect(mockOnClearFilter).toHaveBeenCalledTimes(0);
        fireEvent.click(link);
        expect(mockOnClearFilter).toHaveBeenCalledTimes(1);
      }
    }
  });
});
