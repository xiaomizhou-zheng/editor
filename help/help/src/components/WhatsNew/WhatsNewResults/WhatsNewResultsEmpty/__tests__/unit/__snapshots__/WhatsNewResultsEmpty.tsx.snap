// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhatsNewResultsEmpty Should match snapshot 1`] = `
.emotion-0 {
  padding: 24px 24px 0 24px;
  text-align: center;
}

<div
  class="emotion-0"
>
  <span>
    <!--?xml version="1.0" encoding="UTF-8"?-->
    

    <svg
      fill="none"
      height="116"
      viewBox="0 0 100 116"
      width="100"
      xmlns="http://www.w3.org/2000/svg"
    >
      

      <path
        d="M0 0H100V116H0V0Z"
      />
      

      <g
        opacity="0.3"
      >
        

        <path
          clip-rule="evenodd"
          d="M14.4783 115.563H23.4169H23.4237C24.2045 115.563 24.8375 114.928 24.8375 114.144C24.8375 113.36 24.2045 112.725 23.4237 112.725H14.4783C13.6975 112.725 13.0645 113.36 13.0645 114.144C13.0645 114.928 13.6975 115.563 14.4783 115.563ZM40.6686 115.563H31.7301C30.9492 115.563 30.3162 114.928 30.3162 114.144C30.3162 113.36 30.9492 112.725 31.7301 112.725H40.6686C41.4495 112.725 42.0825 113.36 42.0825 114.144C42.0825 114.928 41.4495 115.563 40.6686 115.563ZM48.975 115.563H57.9204C58.7013 115.563 59.3343 114.928 59.3343 114.144C59.3343 113.36 58.7013 112.725 57.9204 112.725H48.975C48.1942 112.725 47.5612 113.36 47.5612 114.144C47.5612 114.928 48.1942 115.563 48.975 115.563ZM66.2268 115.563H75.1722C75.953 115.563 76.586 114.928 76.586 114.144C76.586 113.36 75.953 112.725 75.1722 112.725H66.2268C65.446 112.725 64.813 113.36 64.813 114.144C64.813 114.928 65.446 115.563 66.2268 115.563Z"
          fill="#B3BAC5"
          fill-rule="evenodd"
        />
        

        <path
          d="M6.17203 115.563H1.41386C0.633005 115.563 0 114.928 0 114.144V109.368C0 108.584 0.633005 107.948 1.41386 107.948C2.19471 107.948 2.82771 108.584 2.82771 109.368V112.725H6.17203C6.95288 112.725 7.58588 113.36 7.58588 114.144C7.58588 114.928 6.95288 115.563 6.17203 115.563Z"
          fill="#B3BAC5"
        />
        

        <path
          clip-rule="evenodd"
          d="M0.411714 26.0299C0.67706 26.2976 1.03771 26.448 1.41387 26.448V26.4412C1.79003 26.4412 2.15068 26.2907 2.41603 26.0231C2.68138 25.7554 2.82954 25.3927 2.82773 25.0151V15.2574C2.82773 14.4736 2.19472 13.8381 1.41387 13.8381C0.633021 13.8381 1.63631e-05 14.4736 1.63631e-05 15.2574V25.0219C-0.00179209 25.3995 0.146369 25.7623 0.411714 26.0299ZM1.41387 45.2673C1.03771 45.2673 0.67706 45.1168 0.411714 44.8492C0.146369 44.5816 -0.00179209 44.2188 1.63631e-05 43.8412V34.0835C1.63631e-05 33.2997 0.633021 32.6643 1.41387 32.6643C2.19472 32.6643 2.82773 33.2997 2.82773 34.0835V43.8412C2.82773 44.625 2.19472 45.2605 1.41387 45.2605V45.2673ZM0.411714 63.6685C0.67706 63.9361 1.03771 64.0866 1.41387 64.0866C1.79003 64.0866 2.15068 63.9361 2.41603 63.6685C2.68138 63.4008 2.82954 63.0381 2.82773 62.6605V52.9028C2.82773 52.119 2.19472 51.4835 1.41387 51.4835C0.633021 51.4835 1.63631e-05 52.119 1.63631e-05 52.9028V62.6605C-0.00179209 63.0381 0.146369 63.4008 0.411714 63.6685ZM1.41387 82.9059C0.633021 82.9059 1.63631e-05 82.2705 1.63631e-05 81.4866V71.729C1.63631e-05 70.9451 0.633021 70.3097 1.41387 70.3097C2.19472 70.3097 2.82773 70.9451 2.82773 71.729V81.4866C2.82773 81.863 2.67877 82.224 2.41362 82.4902C2.14847 82.7564 1.78885 82.9059 1.41387 82.9059ZM1.41387 101.725C0.633021 101.725 1.63631e-05 101.09 1.63631e-05 100.306V90.5482C1.63631e-05 89.7644 0.633021 89.129 1.41387 89.129C2.19472 89.129 2.82773 89.7644 2.82773 90.5482V100.306C2.82773 101.09 2.19472 101.725 1.41387 101.725Z"
          fill="#B3BAC5"
          fill-rule="evenodd"
        />
        

        <path
          d="M1.41386 7.61506C0.633005 7.61506 0 6.97962 0 6.19577V1.41929C0 0.63544 0.633005 0 1.41386 0H6.17203C6.95288 0 7.58588 0.63544 7.58588 1.41929C7.58588 2.20315 6.95288 2.83859 6.17203 2.83859H2.82771V6.19577C2.82771 6.97962 2.19471 7.61506 1.41386 7.61506Z"
          fill="#B3BAC5"
        />
        

        <path
          clip-rule="evenodd"
          d="M14.4783 2.83859H23.4169C24.1977 2.83859 24.8307 2.20315 24.8307 1.41929C24.8307 0.63544 24.1977 0 23.4169 0H14.4783C13.6975 0 13.0645 0.63544 13.0645 1.41929C13.0645 2.20315 13.6975 2.83859 14.4783 2.83859ZM40.6618 2.83859H31.7233C30.9424 2.83859 30.3094 2.20315 30.3094 1.41929C30.3094 0.63544 30.9424 0 31.7233 0H40.6618C41.4427 0 42.0757 0.63544 42.0757 1.41929C42.0757 2.20315 41.4427 2.83859 40.6618 2.83859ZM48.9411 2.83859H57.9068C58.6877 2.83859 59.3207 2.20315 59.3207 1.41929C59.3207 0.63544 58.6877 0 57.9068 0H48.9411C48.1602 0 47.5272 0.63544 47.5272 1.41929C47.5272 2.20315 48.1602 2.83859 48.9411 2.83859ZM66.2132 2.83859H75.1586C75.9394 2.83859 76.5724 2.20315 76.5724 1.41929C76.5724 0.63544 75.9394 0 75.1586 0H66.2132C65.4324 0 64.7994 0.63544 64.7994 1.41929C64.7994 2.20315 65.4324 2.83859 66.2132 2.83859Z"
          fill="#B3BAC5"
          fill-rule="evenodd"
        />
        

        <path
          d="M88.2169 7.61506C87.4361 7.61506 86.8031 6.97962 86.8031 6.19577V2.83859H83.4588C82.6779 2.83859 82.0449 2.20315 82.0449 1.41929C82.0449 0.63544 82.6779 0 83.4588 0H88.2169C88.5919 0 88.9515 0.149532 89.2167 0.415702C89.4818 0.681871 89.6308 1.04287 89.6308 1.41929V6.19577C89.6308 6.97962 88.9978 7.61506 88.2169 7.61506Z"
          fill="#B3BAC5"
        />
        

        <path
          clip-rule="evenodd"
          d="M86.8164 25.0287C86.8164 25.8126 87.4494 26.448 88.2303 26.448L88.2235 26.4412C88.6008 26.443 88.9632 26.2934 89.23 26.0255C89.4968 25.7577 89.6459 25.3939 89.6441 25.0151V15.2574C89.6441 14.4736 89.0111 13.8381 88.2303 13.8381C87.4494 13.8381 86.8164 14.4736 86.8164 15.2574V25.0287ZM88.2303 45.2673C87.4494 45.2673 86.8164 44.6319 86.8164 43.848V34.0835C86.8164 33.2997 87.4494 32.6643 88.2303 32.6643C89.0111 32.6643 89.6441 33.2997 89.6441 34.0835V43.8412C89.6441 44.2188 89.4942 44.5808 89.2276 44.8472C88.961 45.1136 88.5996 45.2623 88.2235 45.2605L88.2303 45.2673ZM86.8164 62.6673C86.8164 63.4512 87.4494 64.0866 88.2303 64.0866L88.2235 64.0798C88.5996 64.0816 88.961 63.9329 89.2276 63.6665C89.4942 63.4001 89.6441 63.0381 89.6441 62.6605V52.9028C89.6441 52.119 89.0111 51.4835 88.2303 51.4835C87.4494 51.4835 86.8164 52.119 86.8164 52.9028V62.6673ZM86.8164 81.4866C86.8164 82.2705 87.4494 82.9059 88.2303 82.9059C88.6052 82.9059 88.9649 82.7564 89.23 82.4902C89.4952 82.224 89.6441 81.863 89.6441 81.4866V71.729C89.6441 70.9451 89.0111 70.3097 88.2303 70.3097C87.4494 70.3097 86.8164 70.9451 86.8164 71.729V81.4866Z"
          fill="#B3BAC5"
          fill-rule="evenodd"
        />
        

      </g>
      

      <path
        d="M73.1399 80.1628L69.0207 76.0687L63.3721 81.8141L67.4913 85.9083C68.542 86.9484 69.2773 88.2662 69.6121 89.709C69.9468 91.1517 70.6822 92.4695 71.7329 93.5097L88.434 110.064C90.9645 112.569 95.039 112.542 97.5357 110.002C100.032 107.462 100.004 103.372 97.4746 100.865L80.7734 84.3252C79.7179 83.2947 78.3935 82.585 76.9532 82.2781C75.5079 81.9508 74.1852 81.2171 73.1399 80.1628Z"
        fill="#CFD4DB"
      />
      

      <path
        d="M73.1398 80.1628L71.889 78.9209C70.3065 77.3556 67.7597 77.3739 66.1996 78.9618C64.6403 80.5505 64.6586 83.1071 66.2404 84.6731L67.4911 85.915C68.5418 86.9551 69.2772 88.273 69.6119 89.7157C69.9466 91.1584 70.682 92.4763 71.7327 93.5164L88.4339 110.063C90.9644 112.569 95.0389 112.541 97.5356 110.002C100.031 107.462 100.004 103.372 97.4744 100.865L80.7732 84.3251C79.7177 83.2947 78.3934 82.585 76.9531 82.2781C75.5077 81.9507 74.185 81.217 73.1398 80.1628Z"
        fill="url(#paint0_linear)"
      />
      

      <g
        style="mix-blend-mode:multiply"
      >
        

        <path
          d="M76.8105 82.1963C75.4309 81.8599 74.1689 81.1522 73.1603 80.1492L71.9096 78.9073C70.327 77.342 67.7802 77.3604 66.2201 78.9483C64.6608 80.5369 64.6791 83.0935 66.2609 84.6596L67.5116 85.9015C68.5178 86.9004 69.2345 88.1552 69.5848 89.5316C72.3503 87.4687 74.7867 84.9953 76.8105 82.1963Z"
          fill="#DFE1E5"
        />
        

      </g>
      

      <path
        clip-rule="evenodd"
        d="M11.6101 59.4193C11.734 78.0735 26.8988 93.0949 45.4815 92.9706V92.9501C59.0855 92.8572 71.2946 84.5482 76.417 71.8966C81.5395 59.2451 78.5668 44.7418 68.8849 35.148C62.5481 28.8366 53.9591 25.3328 45.0329 25.4176C26.4501 25.542 11.4863 40.7651 11.6101 59.4193ZM24.0596 73.631C17.2047 63.4768 18.4498 49.8758 27.0334 41.1459C31.7943 36.277 38.2978 33.5226 45.0941 33.4967C57.313 33.4162 67.8869 42.0115 70.3492 54.0261C72.8116 66.0406 66.4765 78.1281 55.2183 82.8962C43.9601 87.6643 30.9145 83.7851 24.0596 73.631Z"
        fill="url(#paint1_linear)"
        fill-rule="evenodd"
      />
      

      <path
        d="M57.2614 50.9444L53.754 47.3962C53.2279 46.8893 52.3972 46.8893 51.8711 47.3962L45.2776 53.9673L38.7453 47.3416C38.2193 46.8347 37.3885 46.8347 36.8625 47.3416L33.3074 50.8557C32.8025 51.3838 32.8025 52.2178 33.3074 52.7458L39.8669 59.3647L33.3074 65.9289C32.8025 66.457 32.8025 67.2909 33.3074 67.819L36.8149 71.3673C37.3409 71.8742 38.1717 71.8742 38.6978 71.3673L45.2504 64.8235L51.7963 71.4423C52.3224 71.9492 53.1532 71.9492 53.6792 71.4423L57.2138 67.9214C57.7188 67.3933 57.7188 66.5593 57.2138 66.0313L50.668 59.3647L57.2614 52.7936C57.7408 52.2712 57.7408 51.4669 57.2614 50.9444Z"
        fill="#C1C7D0"
      />
      

      <defs>
        

        <lineargradient
          gradientUnits="userSpaceOnUse"
          id="paint0_linear"
          x1="64.6666"
          x2="99.3101"
          y1="111.866"
          y2="112.308"
        >
          

          <stop
            stop-color="#B3BAC5"
          />
          

          <stop
            offset="1"
            stop-color="#DFE1E5"
            stop-opacity="0.5"
          />
          

        </lineargradient>
        

        <lineargradient
          gradientUnits="userSpaceOnUse"
          id="paint1_linear"
          x1="44.6642"
          x2="93.0149"
          y1="106.955"
          y2="59.995"
        >
          

          <stop
            offset="0.56"
            stop-color="#C1C7D0"
          />
          

          <stop
            offset="0.97"
            stop-color="#E9EBEF"
            stop-opacity="0.5"
          />
          

        </lineargradient>
        

      </defs>
      

    </svg>
    

  </span>
</div>
`;
